avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -500,6 +503,27 @@ private static Object fromConnectData(Schema schema, org.apache.avro.Schema avro
     }
   }
 
+  private static org.apache.avro.Schema avroSchemaForUnderlyingMapEntryType(
+      Schema schema,
+      org.apache.avro.Schema avroSchema
+  ) {
+    if (schema != null && schema.isOptional()) {
+      if (avroSchema.getType() == org.apache.avro.Schema.Type.UNION) {
+        for (org.apache.avro.Schema typeSchema : avroSchema.getTypes()) {
+          if (!typeSchema.getType().equals(org.apache.avro.Schema.Type.NULL) &&
+              typeSchema.getFullName().equals("array")) {
```

Comment:
Would be good to use a constant for "array". Also, is there any chance that `typeSchema.getFullName()` might be null? If so, the comparison could be reversed:

    "array".equals(typeSchema.getFullName())